lib/_io.scm

```diff
@@ -8343,6 +8343,167 @@
         (macro-dynamic-bind output-port port thunk)))))
 
 ;;;----------------------------------------------------------------------------
+(define-prim (##open-raw-device
+              direction
+              name
+              fd)
+  (define (fail)
+    (##fail-check-settings 1 ##open-raw-device direction name fd))
+
+  (##make-psettings
+   direction
+   '()
+   '()
+   fail
+   (lambda (psettings)
+     (let ((device
+            (##os-device-open-raw
```

Comment:
So I am not clear why the caller should be doing the `##os-device-open-raw` here.
The presumption is that the made a device handle and now wants to open a raw device.
We either should drop the function altogether (and have user construct psettings) or keep it as is.